src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorDiagnosticsEndpoint.cs

```diff
@@ -93,22 +73,9 @@ public async Task<RazorDiagnosticsResponse> Handle(RazorDiagnosticsParams reques
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            int? documentVersion = null;
-            DocumentSnapshot? documentSnapshot = null;
-            await _projectSnapshotManagerDispatcher.RunOnDispatcherThreadAsync(() =>
-            {
-                _documentResolver.TryResolveDocument(request.RazorDocumentUri.GetAbsoluteOrUNCPath(), out documentSnapshot);
-
-                Debug.Assert(documentSnapshot != null, "Failed to get the document snapshot, could not map to document ranges.");
-
-                if (documentSnapshot is null ||
-                    !_documentVersionCache.TryGetDocumentVersion(documentSnapshot, out documentVersion))
-                {
-                    documentVersion = null;
```

Comment:
This ends up being a functional change (it fails a test), though it is one that AFAIK should never be the case in regular usage. I modified the expectations of the test in question, but this area bears extra scrutiny since the contract has technically changed.